File: examples/gin_demo/apps/note/table.go
package note

import (
    "github.com/google/uuid"
)

type Note struct {
    ID        uuid.UUID `gorm:"type:uuid;default:uuid_generate_v4();primaryKey" json:"id"`
    Title     string    `gorm:"size:255;not null" json:"title"`
    Content   string    `gorm:"type:text;not null" json:"content"`
}

File: examples/fastapi_demo/apps/note/table.py
import uuid
from sqlmodel import SQLModel, Field
from typing import Optional

class Note(SQLModel, table=True):
    id: Optional[uuid.UUID] = Field(default_factory=uuid.uuid4, primary_key=True)
    title: str = Field(max_length=255)
    content: str

# 输入输出模型
class NoteInsert(SQLModel):
    title: str
    content: str

class NoteUpdate(SQLModel):
    title: Optional[str] = None
    content: Optional[str] = None

# Note 类既是数据表模型也是响请求响应模型
